datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -450,6 +466,13 @@ public StickyPartitionAssignmentStrategy(Optional<Integer> maxTasks, Optional<In
     return tasksToCleanUp;
   }
 
+  @Override
+  public void cleanupStrategy() {
+    // Clear the maintained state
+    _elasticTaskAssignmentInfoHashMap.clear();
```

Comment:
You can deregister the gauge metrics as well.

Reply:
donee